core/src/main/java/io/grpc/ServerBuilder.java

```diff
@@ -89,6 +89,20 @@
   public abstract T addService(BindableService bindableService);
 
   /**
+   * Adds a {@link ServerInterceptor} that is run for all services on the server.  Interceptors
+   * added through this method always run before per-service interceptors added through {@link
+   * ServerInterceptors}.  Interceptors run in the reverse order in which they are added.
+   *
+   * @param interceptor the all-service interceptor
+   * @return this
+   * @since 1.5.0
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/3117")
+  public T addInterceptor(ServerInterceptor interceptor) {
```

Comment:
While I wouldn't disagree that `addInterceptor()` may be a better name, it is probably better to copy `ManagedChannelBuilder`'s `intercept()` name.